pkg/register/register.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package register
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"github.com/rancher/elemental-operator/pkg/dmidecode"
+	"github.com/rancher/elemental-operator/pkg/tpm"
+	"github.com/sirupsen/logrus"
+)
+
+func Register(url string, caCert []byte, smbios bool, emulatedTPM bool, emulatedSeed int64, labels map[string]string) ([]byte, error) {
+
+	dialer := websocket.DefaultDialer
+
+	if len(caCert) > 0 {
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+		pool.AppendCertsFromPEM(caCert)
+		dialer.TLSClientConfig = &tls.Config{RootCAs: pool}
+	}
+
+	tpmAuth := &tpm.AuthClient{}
+	if emulatedTPM {
+		tpmAuth.EmulateTPM(emulatedSeed)
```

Comment:
would add info saying that we are using emu tpm :)

Reply:
thanks, done!